File: April 24, 2026.js
function compress(str) {

  const wordsList = {};
  let newStr = {};
  let result = [];
  let occurrence = 0;

  for (let word of str.split(" ")) {
    occurrence += 1;
    if (!Object.hasOwn(wordsList, word)) {
      wordsList[word] = occurrence;
      result.push(word);
    } else {
      result.push(String(wordsList[word]));
    }
  }

  newStr = result.join(" ");

  return newStr;
}

File: April 24, 2026.py
def compress(s):

    words_list = {}
    new_s = ""
    result = []
    occurrence = 0

    for word in s.split(" "):
        occurrence += 1
        if word not in words_list:
            words_list[word] = occurrence
            result.append(word)
        else:
            result.append(str(words_list[word]))

    new_s = " ".join(result)

    return new_s